### ISSUE TITLE
Want to see your loss curve...

### ISSUE THREAD

**Issue by JarveeLee** [ZJULearning/pixel_link#38]
I am trying to implement this paper under pytorch.
It has been done except loss weight , wish to achieve your performance in Table5 , 3 "without instance balance "
But my loss value keep wondering at a high level for hours.....and behaving like a pure FCN in Table5,2 , predicting links all of 1 with neighbor pixels....
So could you show your loss curve or train log ? or any suggestion ?

**Comment by mattroos** [comment]
I'm not affiliated with the authors, but I did run their training code and plot the loss curve. I also did a training run that started with their pretrained model. Both are for the "PixelLink + VGG16 2s" model.

### Here is what I got when training from scratch.
The loss was still decreasing when I stopped it, but was at about 0.76. 

*Some caveats:*
Deng et al. trained with 3 GPUs, 24 images/GPU, for 72 images/batch. To my understanding, the batch size is equivalent to one "step" in the code, also called one "iteration" in the paper. I trained with 1 GPU and 4 images/GPU, for a batch/step/interation size of 4 images. I only trained with a learning rate of 1e-3 (The authors did initial training with LR=1e-3 then changed to LR=1e-2. The loss blew up when I tried that. It might have just had a poor initialization.)
![Figure_1](https://user-images.githubusercontent.com/7317974/59881485-b6e27400-937d-11e9-8ea9-8101eb62855d.png)

### Here is what I got when loading the pretrained model, then training further.

The curve is noisy because I only ran it for a short while, so each marker on the plot represents an average of many fewer steps than in the plot above.

*Note that:*
Training starts at 73108 steps (the number of training steps in the pretrained model), for the authors' step size of 72 images/batch.
Here I used LR=1e-2, which was simply because I forgot to change it to 1e-3.
The curve is noisy in part because I only trained for a short while, so each marker in the plot represents the average of many fewer steps (and samples) than in the plot above.
The loss continues to decrease. This may be due to over-fitting. The model is large, the training set relatively small, and we are observing losses from the training set, not the test set.
![Figure_2](https://user-images.githubusercontent.com/7317974/59881623-09239500-937e-11e9-8a92-a0985d49228f.png)

**Comment by mattroos** [comment]
@dengdan, I suggest closing this.
